### ISSUE TITLE
Error: Cannot find module './match'

### ISSUE THREAD

**Issue by phourus** [rackt/react-router#3026]
When deploying to AWS ElasticBeanstalk, Node 0.12.2, React-Router 1.0.3, I am getting the following error when starting the server:

Error: Cannot find module './match'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/var/app/current/node_modules/react-router/lib/index.js:96:15)
    at Module._compile (module.js:460:26)
    at Module._extensions..js (module.js:478:10)
    at Object.require.extensions.(anonymous function) [as .js] (/var/app/current/node_modules/babel/node_modules/babel-core/lib/api/register/node.js:214:7)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)

Interestingly enough, I do not encounter this issue locally when using NVM 0.12.2. However, it does seem like the require('./match') is case-sensitive on my AWS instance (but not OX X) and is therefore not finding lib/Match.js because it is looking for lib/match.js

Seems pretty minor but definitely causing some headaches over here!

**Comment by taion** [comment]
Thanks for your question!

We want to make sure that the GitHub issue tracker remains the best place to track bug reports and feature requests that affect the development of React Router.

Questions like yours deserve a purpose-built Q&A forum. Could you post this question to Stack Overflow with the tag #react-router? https://stackoverflow.com/questions/ask?tags=react-router

We also have an active and helpful React Router community on Reactiflux, which is a great place to get fast help with React Router and with the rest of the React ecosystem. You can join at https://discord.gg/0ZcbPKXt5bYaNQ46.

**Comment by phourus** [comment]
UPDATE: This is indeed a bug with v1.0.3, on Linux. Linux being case-sensitive, unlike OS X, causes an error when trying to require './match' because the file name is Match.js. 

Renaming Match.js to match.js should fix this problem on Linux.

This is especially problematic on ElasticBeanstalk with AWS, which by default re-installs all node modules for each deployment.

**Comment by taion** [comment]
There's something messed up in your environment. It is in fact `package/lib/match.js` and `package/es6/match.js` in the tarball, with that exact casing.

**Comment by phourus** [comment]
<img width="300" alt="screen shot 2016-02-04 at 6 02 55 pm" src="https://cloud.githubusercontent.com/assets/3272537/12835997/b3b03c02-cb69-11e5-862a-bb62ed90d85a.png">
<img width="178" alt="screen shot 2016-02-04 at 6 03 08 pm" src="https://cloud.githubusercontent.com/assets/3272537/12835996/b3af7696-cb69-11e5-8bb9-f8311fd7f313.png">
<img width="164" alt="screen shot 2016-02-04 at 6 03 17 pm" src="https://cloud.githubusercontent.com/assets/3272537/12835998/b3b123c4-cb69-11e5-8f14-5be2255fbf44.png">

When installing via npm it is es6/match.js and lib/Match.js, haven't checked the tarball

**Comment by taion** [comment]
Again, I'd suggest double-checking your deployment. Unless you think that you are literally the only person in the world running React Router on Linux (hint: you're not), this is an error on your side. Please reach out elsewhere for support – I've attached a number of links upthread.

**Comment by phourus** [comment]
After hoping that 2.0 would just put this to rest, I am still having an issue and there has to be something wrong with the package. After running npm install locally on OS X I get lib/match.js, when npm install is running on Linux (AWS ElasticBeanstalk) it is lib/Match.js and triggers an error when running npm start, saying the module could not be found.

After terminating the environment, rebuilding and even deploying a new test environment I am still getting the same behavior. I SSHed into the instance, simply change index.js to require('./Match') instead of './match' and it works. 

What I am most confused by is why I'm getting different filenames in different environments after confirming both are react-router 2.0.

Truly baffled by this one, and is a major problem due to the inability to deploy without manually SSHing into my instance and change the require statement. Any ideas what could be happening?

<img width="489" alt="screen shot 2016-02-10 at 1 56 51 pm" src="https://cloud.githubusercontent.com/assets/3272537/12963353/e7e5dccc-cffe-11e5-8839-7fe0fe2bedb8.png">
<img width="190" alt="screen shot 2016-02-10 at 1 56 59 pm" src="https://cloud.githubusercontent.com/assets/3272537/12963354/e7e7adb8-cffe-11e5-99a5-1fb74b3578e0.png">

**Comment by timdorr** [comment]
What version of node/npm do you have installed? You should see the lowercase match.js in https://registry.npmjs.org/react-router/-/react-router-2.0.0.tgz

**Comment by arasmussen** [comment]
I'm also getting this same issue on AWS EC2 (linux) where match.js is located at node_modules/react-router/lib/Match.js. I'm not really sure how or why. One quirky thing I've done is I installed the module on OS X, committed it to my git repo, then pulled it on my linux server. This is probably not the recommended way of doing things.

I did the following and this solved my issue:

```bash
rm -rf node_modules/react-router
npm install
```
